### PR TITLE
LITE-27196 product lookup transformation

### PR DESCRIPTION
**Pull request by Sainomori**

### CODE REVIEW

File: connect_transformations/lookup_product_items/mixins.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        try:
+            product = await retrieve_product(
+                self.installation_client,
+                self._cache,
+                self._cache_lock,
+                product_id,
+                leave_empty,
+            )
+            if (not product) and leave_empty:
```
Comment (ffaraone) [inline connect_transformations/lookup_product_items/mixins.py:47]: These parenthesis are redundant.

File: connect_transformations/lookup_product_items/utils.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+from connect.client.exceptions import ClientError
+from connect.eaas.core.responses import RowTransformationResponse
+from fastapi.responses import JSONResponse
+
+from connect_transformations.constants import PRODUCT_ITEM_LOOKUP
+from connect_transformations.exceptions import ProductLookupError
+
+
+async def retrieve_product(client, cache, cache_lock, product_id, leave_empty):
+    cache_key = f'product-{product_id}'
```
Comment (ffaraone) [inline connect_transformations/lookup_product_items/utils.py:15]: no need to prefix the key with `product-`

File: connect_transformations/lookup_product_items/utils.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+async def _get_product_item_by_id(client, product, lookup_value):
+    try:
+        return await client.products[product['id']].items[lookup_value].get()
+    except ClientError:
+        return None
+
+
+async def _get_product_item_by_filter(client, product, lookup_value):
+    filter_expression = f"ilike(mpn,*{lookup_value}*)"
```
Comment (ffaraone) [inline connect_transformations/lookup_product_items/utils.py:39]: Are we sure that we should use ilike here and not a strict equality ?

**Review by Sainomori — COMMENTED** [review]
It seems like I didn't break anything!

File: ui/src/pages/transformations/lookup_product_item.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      document.getElementById('criteria').appendChild(option);
+    });
+
+    availableColumns.forEach((column) => {
+      const option = document.createElement('option');
+      option.value = column.id;
+      option.text = column.name;
+      document.getElementById('column').appendChild(option);
+
+      const anotherOption = document.createElement('option');
```
Comment (veralimita) [inline ui/src/pages/transformations/lookup_product_item.js:58]: Why you did not append ```option``` twice?

Comment (Sainomori) [reply ui/src/pages/transformations/lookup_product_item.js:58]: Tried. It appends only in the first select. In the second one, an empty option is appended.

**Pull request merged**